File: 2018/src/test/java/com/benjamin/Day3Test.java
package com.benjamin;

import org.hamcrest.Matchers;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.hamcrest.MatcherAssert.assertThat;

public class Day3Test {

    private Day3 target;
    private static final String INPUT1 =
            "#1 @ 1,3: 4x4\n" +
            "#2 @ 3,1: 4x4\n" +
            "#3 @ 5,5: 2x2\n";

    @BeforeEach
    public void setup() {
        target = new Day3();
    }

    @Test
    public void testDeelEen1() {
        int result = target.deelEenA(INPUT1);

        assertThat(result, Matchers.is(4));
    }

    @Test
    public void testDeelTwee1() {
        int result = target.deelTweeA(INPUT1);

        assertThat(result, Matchers.is(-1));
    }
}

File: 2021/src/test/kotlin/Day3Test.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

internal class Day3Test {

  private val testInput = """00100
11110
10110
10111
10101
01111
00111
11100
10000
11001
00010
01010"""

  @Test
  fun `should test part1`() {
    val testInput = testInput

    assertEquals(198, Day3.part1(testInput))
  }

  @Test
  fun `should test part2`() {
    val testInput = testInput

    assertEquals(230, Day3.part2(testInput))
  }
}
